File: c++/1.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;

vector<int> firstColumn;
vector<int> lastColumn;

void readInput() {
    ifstream input{"../../data/1.txt"};

    string line;
    while (getline(input, line)) {
        istringstream iss(line);
        int first, second;
        iss >> first >> second;

        firstColumn.push_back(first);
        lastColumn.push_back(second);
    }
}

int totalDistance() {
    int total = 0;

    for (int i = 0; i < firstColumn.size(); i++) {
        total += abs(lastColumn[i] - firstColumn[i]);
    }

    return total;
}

int similarity() {
    int total = 0;

    for (const int x: firstColumn) {
        total += x * (int) ranges::count(lastColumn, x);
    }

    return total;
}

int main() {
    readInput();
    ranges::sort(firstColumn);
    ranges::sort(lastColumn);

    cout << totalDistance() << endl;
    cout << similarity() << endl;

    return 0;
}

File: python/1.py
lines = open("../data/1.txt").read().splitlines()
first_column, last_column = [], []

for line in lines:
    x, y = line.split()
    first_column.append(int(x))
    last_column.append(int(y))

first_column.sort()
last_column.sort()

total = 0
for i in range(len(first_column)):
    total += abs(first_column[i] - last_column[i])

similarity = 0
for x in first_column:
    similarity += x * last_column.count(x)

print(total)
print(similarity)